RELEASE.rst

```diff
@@ -1,6 +1,24 @@
 Release Notes
 -------------
 
+Version 0.3.0
+=============
+
+- Added UI to add and remove repository members.
+- Added form for adding new vocabularies.
+- Added manage taxonomies panel and button
+- Tests for members REST.
```

Comment:
I think we should drop this item from the release notes. It's part of the next item, "REST API for repo members"